csharp/Api/BUILD

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (C) 2022 Vaticle
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+package(default_visibility = ["//visibility:public"])
+
+
+load("//csharp:build_opts.bzl", "nullable_context", "target_frameworks", "targeting_packs")
+load("@rules_dotnet//dotnet:defs.bzl", "csharp_library")
+load("@vaticle_dependencies//tool/checkstyle:rules.bzl", "checkstyle_test")
+
+
+csharp_library(
+    name = "api",
```

Comment:
Should the target names follow the package names and also be named in `UpperCamelCase`?

Reply:
It is actually a good point! 

However, it looks like (if we want to follow it...) it's even worse: following the [guideline](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-assemblies-and-dlls), our targets (which are dlls) should be like `Typedb.Driver.***` as well

Reply:
Could be especially confusing in terms of tests: `Typedb.Driver.Behaviour.Connection.Database.Steps`, `Typedb.Driver.Behaviour.Connection.Database.TestCore`... 

Not sure if we want, although it would be better to make user-facing libs following the convensions.

Reply:
It looks like I can't specify a separate name for the library file using [the rule](https://github.com/bazelbuild/rules_dotnet/blob/master/docs/csharp_library.md) (`[impl](https://github.com/bazelbuild/rules_dotnet/blob/862c7f5b8bc9d13de064ef0183e30da11034f3a4/dotnet/private/rules/csharp/library.bzl#L4)`). Moreover, these files are generated without namespaces/packages in their names. so its just `driver-csharp.dll`, `api.dll`, `common.dll` for now. So I'm not sure what to follow and if we want to try to rename these files with an additional rule to have `TypeDB.Driver.Api`and... Uhm... `TypeDB.Driver`?